validation.go

```diff
@@ -222,46 +226,61 @@ func validateParamValue(key string, value string, param Parameter) error {
 	return multiErr
 }
 
-// assignParamDefaults fills any empty configuration with its assigned default value
+// initConfig checks for unrecognized params, and fills any empty configuration with its assigned default value
 // returns an error if a parameter is not recognized.
-func assignParamDefaults(params map[string]Parameter, config map[string]string) (map[string]string, error) {
+func (v validator) initConfig(config map[string]string) (map[string]string, error) {
 	output := make(map[string]string)
 	var multiErr error
 	for key, val := range config {
-		if _, ok := params[key]; !ok {
+		// trim key and val
+		key = strings.TrimSpace(key)
+		val = strings.TrimSpace(val)
+		// check for unrecognized params
+		if _, ok := v[key]; !ok {
 			multiErr = multierr.Append(multiErr, fmt.Errorf("unrecognized parameter %q", key))
 			continue
 		}
+		// set config map and assign defaults
 		output[key] = val
-		if len(strings.TrimSpace(val)) == 0 {
-			output[key] = params[key].Default
+		if val == "" {
+			output[key] = v[key].Default
+		}
+	}
+	// set defaults for unassigned params
+	for key, param := range v {
+		if _, ok := output[key]; !ok {
```

Comment:
... and change this to check if the value is empty.
```suggestion
		if output[key] == "" {
```

Reply:
wow that was something I wanted to do but forgot to for some reason 🥲 thanks for the suggestion